From a1abd14e6bf634610fe78fca245303186c58e821 Mon Sep 17 00:00:00 2001
From: Wiktor Ciurej <wiktor.ciurej@gmail.com>
Date: Sat, 4 Apr 2020 15:23:04 +0200
Subject: [PATCH] Fix build after y2038 changes in glibc

---
 src/dhcp.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/dhcp.c b/src/dhcp.c
index 4a2983e..9719a27 100644
--- a/src/dhcp.c
+++ b/src/dhcp.c
@@ -18,6 +18,10 @@
 
 #ifdef HAVE_DHCP
 
+#ifndef SIOCGSTAMP_OLD
+#define SIOCGSTAMP_OLD SIOCGSTAMP
+#endif
+
 struct iface_param {
   struct dhcp_context *current;
   struct dhcp_relay *relay;
@@ -179,7 +183,7 @@ void dhcp_packet(time_t now, int pxe_fd)
     return;
     
   #if defined (HAVE_LINUX_NETWORK)
-  if (ioctl(fd, SIOCGSTAMP, &tv) == 0)
+  if (ioctl(fd, SIOCGSTAMP_OLD, &tv) == 0)
     recvtime = tv.tv_sec;
 
   if (msg.msg_controllen >= sizeof(struct cmsghdr))
-- 
2.20.1

