From b45f6344c0ca657180ec908354b9f7406ee0200c Mon Sep 17 00:00:00 2001
From: Wiktor Ciurej <wiktor.ciurej@gmail.com>
Date: Fri, 3 Apr 2020 21:37:30 +0200
Subject: [PATCH 16/16] FROMGIT: Input: add privacy screen toggle keycode

---
 include/uapi/linux/input-event-codes.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/uapi/linux/input-event-codes.h b/include/uapi/linux/input-event-codes.h
index 85387c76c..05d8b4f4f 100644
--- a/include/uapi/linux/input-event-codes.h
+++ b/include/uapi/linux/input-event-codes.h
@@ -649,6 +649,8 @@
  */
 #define KEY_DATA			0x277
 #define KEY_ONSCREEN_KEYBOARD		0x278
+/* Electronic privacy screen control */
+#define KEY_PRIVACY_SCREEN_TOGGLE	0x279
 
 #define BTN_TRIGGER_HAPPY		0x2c0
 #define BTN_TRIGGER_HAPPY1		0x2c0
-- 
2.20.1

